docs: addes subject and audit for the travel-plan project

zanninso opened the pull request.

nprimo commented on subjects/java/projects/travel-plan/README.md:
```diff
@@ -0,0 +1,41 @@
+
+### Instructions
+
+#### 1. Environment Configuration
```
It would be nice to specify what type of files/structure is expected from the students

nprimo commented on subjects/java/projects/travel-plan/README.md:
```diff
@@ -0,0 +1,41 @@
+
+#### 1. Environment Configuration
+
+In the initial phase of the Travel Management System, students will configure a microservices architecture following industry best practices to ensure scalability and high availability. This includes implementing multiple replicas of each microservice to support load balancing and failover mechanisms, essential for achieving system resilience and service continuity. The task encompasses the insta ...
```
It would be great to have bullet points or more specific requirements for the task. For example:
- what's the minimum number of microservices expected? Should there be one Ansible playbook per microservice defined?
- what are the functionalities required from the CI/CD pipeline? Should it be running some unit test, a linter?

nprimo commented on subjects/java/projects/travel-plan/README.md:
```diff
@@ -0,0 +1,41 @@
+
+- Craft an Admin Dashboard enabling administrators to oversee users, travels, and payment gateways.
+- Ensure each travel entry includes a destination or trajectory, dates, duration, activities or itinerary, accommodation, and transportation details.
+- Develop an authentication and authorization service to safeguard access and operations.
```
I would specify (I might have misunderstood) that there should be at least two level of authorization: users and admins

nprimo commented on subjects/java/projects/travel-plan/README.md:
```diff
@@ -0,0 +1,41 @@
+
+In the initial phase of the Travel Management System, students will configure a microservices architecture following industry best practices to ensure scalability and high availability. This includes implementing multiple replicas of each microservice to support load balancing and failover mechanisms, essential for achieving system resilience and service continuity. The task encompasses the insta ...
+
+#### 2. Development and Design
```
I would better clarify the functionalities of the application: what is supposed to be developed is quite vague for me.

From what I understood, I would add something like this:

```
The platform is supposed to have 2 main users: "normal users" and "admins". Users must be able to:
- register and login 
- add and manage their travels, including ... 
- (add all other features requested)...

On the other hand, admins must be able to:
- register and login 
- oversee users, travels, and payment gateways (what do you mean by this? - what are the payment gateways supposed to show/enable/...??)
- (other features if any)...

nprimo replied:
I would specify if only the admin part is the one supposed to be devloped

nprimo commented on subjects/java/projects/travel-plan/README.md:
```diff
@@ -0,0 +1,41 @@
+- Craft an Admin Dashboard enabling administrators to oversee users, travels, and payment gateways.
+- Ensure each travel entry includes a destination or trajectory, dates, duration, activities or itinerary, accommodation, and transportation details.
+- Develop an authentication and authorization service to safeguard access and operations.
+- Design a responsive and intuitive UI for the Admin Dashboard, ensuring cross-device and browser compatibility.
```
Ensuring cross-device means the platform must be well designed for PC and mobile? I would specify this.

Additionally, is it only for the `Admin Dashboard` or for the whole application?

nprimo commented on subjects/java/projects/travel-plan/audit/README.md:
```diff
@@ -0,0 +1,76 @@
+
+#### Comprehension
+
+##### Describe the microservices architecture implemented.
```
To be inline with the rest of the questions, I would change `Describe..` with `Ask the student to describe...`

nprimo commented on subjects/java/projects/travel-plan/audit/README.md:
```diff
@@ -0,0 +1,76 @@
+
+##### Discuss the CI/CD pipeline setup.
+
+###### Was the CI/CD pipeline, integrating Jenkins and SonarQube, effectively used to improve the development process?
```
I would add some details on how to define if the CI/CD is effective: do we mean it is running unit tests, E2E tests, integration tests, linters ? How often it is triggered, when ....?

nprimo commented on subjects/java/projects/travel-plan/audit/README.md:
```diff
@@ -0,0 +1,76 @@
+
+###### Were comprehensive security measures like SSL/TLS, secret management, and principle of least privilege correctly implemented?
+
+##### Explain the database schema for PostgreSQL and Neo4j.
```
I would change `Explain` with `Ask the student to explain ...`

nprimo commented on subjects/java/projects/travel-plan/audit/README.md:
```diff
@@ -0,0 +1,76 @@
+
+#### Functional
+
+##### Verify the execution of Ansible playbooks.
```
Maybe add which commands is required to use to execute an `Ansible` playbook and what is the expected result

nprimo commented on subjects/java/projects/travel-plan/audit/README.md:
```diff
@@ -0,0 +1,76 @@
+
+###### Did the Ansible playbooks execute without errors and configure the environment as intended?
+
+###### Were the playbooks able to handle re-running scenarios without causing disruptions or inconsistencies?
```
How can you trigger a "re-running scenario"? I would add this information to ease the audit process

nprimo replied:
`Is there a definition of how to "recover" if something bad happens in the playbook`

nprimo replied:
This point is still missing

zanninso replied:
by creating an error in the environment they can  can trigger a re-run. i don't really see that i have to add that both of the student will work on the project and the auditor can ask the student to trigger it in case he doesn't know.
there is multiple ways or scenarios to recover.

nprimo commented on subjects/java/projects/travel-plan/audit/README.md:
```diff
@@ -0,0 +1,76 @@
+
+###### Were the playbooks able to handle re-running scenarios without causing disruptions or inconsistencies?
+
+##### Test each microservice API.
```
I would add more information here: what are the minimum APIs expected or what are the minimum type of interactions expected? For example:
- there must be an endpoint to add a travel and it must be possible to use it only with the proper authentication system enabled
- there must be an endpoint to retrieve all the travel for the logged user 
- there must be an endpoint to retrieve all the travel for all the users, only accessible for admins 
- (all the other minimum endpoints required from the application)

nprimo commented on subjects/java/projects/travel-plan/audit/README.md:
```diff
@@ -0,0 +1,76 @@
+
+##### Verify Admin Dashboard functionality.
+
+###### Did the Admin Dashboard correctly manage users, travels, and payment gateways?
```
What do we mean by "correctly manage users, travellers and payment gateways"? 
- Which type of actions can we do from the admin dashboard? Retrieve information, approve or change information ...?
This links to the comment I made on the subject: I feel we should be clearer on what the functionalities are expected from the student's final application - once defined in the subject, they should be either checked here one by one or there should be something like: `Did all the `Admin Dashboard` features defined in the subject work as expected?`

nprimo replied:
Admins should be able to perform CRUD operations for users, travels and payment gateways?

zanninso replied:
> Admins should be able to perform CRUD operations for users, travels and payment gateways?

yes

nprimo replied:
Please, add this information in the audit question. It could be nice to add, for example:
```
Admin should be able to perform CRUD operations for users, travels and payment methods.
For each "entity" (user, travel, payment method) try to create, read, update and delete.

Is everything working as expected?
Are errors handled correctly?
```

nprimo commented on subjects/java/projects/travel-plan/audit/README.md:
```diff
@@ -0,0 +1,76 @@
+
+###### Was the authentication service robust and did the role-based access control function correctly?
+
+##### Simulate load on microservices.
```
I feel we should provide an example of how to do so

nprimo replied:
Or another option could be to ask to see if there is some specification in the configuration to say when and how the architecture should scale

nprimo commented on subjects/java/projects/travel-plan/audit/README.md:
```diff
@@ -0,0 +1,76 @@
+
+##### Verify Docker and Ansible setup.
+
+###### Were Docker containers and Ansible playbooks set up correctly and functionally?
```
What do we mean by "correctly and functionally"? I would add more details on how to define this

nprimo replied:
This is still missing - the people reviewing the project should check
- there are all the expected Docker containers running after executing the Ansible playbooks (?)
- there should be an instance for the PostgreSQL DB and the Neo4j DB as well 
- there should be an instance of Nginx or caddy? or the "frontend" is going to receive requests directly?

nprimo commented on subjects/java/projects/travel-plan/audit/README.md:
```diff
@@ -0,0 +1,76 @@
+
+##### Review project documentation.
+
+###### Does the project documentation accurately describe the system architecture, database schemas, and setup instructions?
```
It is not mentioned in the main part of the subject that it is required to create a documentation of the application: from the current version I understood it was a bonus point creating a documentation

nprimo commented on subjects/java/projects/travel-plan/audit/README.md:
```diff
@@ -0,0 +1,76 @@
+
+##### Assess code review and best practices.
+
+###### Were coding standards, security best practices, and overall code quality maintained in recent pull requests?
```
Are there any guidelines to help ensure these points are compliant with what is required from the student?

nprimo reviewed (changes requested):
I left some comments inline on what I think might require some improvements. I am happy to have a quick call if something needs to be discussed more in detail :)

nprimo commented on subjects/java/projects/travel-plan/README.md:
```diff
@@ -0,0 +1,64 @@
+
+##### Outcome:
+-    Achieve an automated, scalable infrastructure.
+-    Prepare the groundwork for Admin Dashboard development and additional features of the Travel Management System.*
```
At the end of this line, there is an extra `*`

nprimo commented on subjects/java/projects/travel-plan/README.md:
```diff
@@ -0,0 +1,64 @@
+
+##### Outcome:
+-    Achieve an automated, scalable infrastructure.
+-    Prepare the groundwork for Admin Dashboard development and additional features of the Travel Management System.*
```
If I understood correctly, one of the outcomes should be to have the functional `Admin Dashboard` working - from this 2 bullet points are not clear to me. I would update the second bullet point to specify that:
```
- ...
- Develop the `Admin Dashboard` that would be one feature of the final project `Travel Management System`
```

zanninso replied:
no this is the environment part so the have to prepare the groundwork for the admin dashboard and not the admin dashboard it self

nprimo replied:
Later on, it is required to develop a responsive UI + some APIs: what do you mean by "groundwork"?

zanninso replied:
here i specify the outcome of the first part of that project that is Environment Configuration.
the second part is  Development and Design where they asked to developer the UI and the APIs

nprimo commented on subjects/java/projects/travel-plan/README.md:
```diff
@@ -0,0 +1,64 @@
+#### 2. Development and Design
+
+- Craft an Admin Dashboard enabling administrators to oversee users, travels, and payment gateways.
+- the admin should be able to add, edit or delete any user, travel, payment method and all the related part correctly (cascading!!!) 
```
What do you mean by `cascading`? Maybe add a link to a general Wikipedia article to avoid confusion about the requirements needed

nprimo commented on subjects/java/projects/travel-plan/README.md:
```diff
@@ -0,0 +1,64 @@
+
+- Craft an Admin Dashboard enabling administrators to oversee users, travels, and payment gateways.
+- the admin should be able to add, edit or delete any user, travel, payment method and all the related part correctly (cascading!!!) 
+- Ensure each travel entry includes a destination or trajectory, dates, duration, activities or itinerary, accommodation, and transportation details.
```
What do you mean by `trajectory`? Should this be an array of coordination of point A and point B
What do you mean by `itinerary`? 

To make this easier to "consume" you could split this bullet point in a sub list with some extra details about each field. For example:
```
- Ensure each travel entry includes:
  - a destination,  in form text (or should this be a coordinate?), or a trajectory (how should this information be saved?)
  - dates (should this be an array of date-times or dates? What is this supposed to look like?)
  - duration, which will be a float reporting the total amount of hours (days/weeks ??) spent in the travel
  - activities ?? or itinerary ?? 
  - accommodation ... 
  - transportation details ...
```

zanninso replied:
yes trajectory mean array of coordination instead of single destination

zanninso replied:
if i give them a lot of details about the fields and table it's like I'm giving them a database diagram but i want them to think and discover what they have to do from the least information

nprimo replied:
I feel implementing a database storing `destination` vs `trajectory` as an array of coordination is quite different in terms of complexity. 

Since this is going to be a part of a bigger project, it would be nice for the student to know if they're going to use the `trajectory` for the future steps of this project or if `destination` would be enough

nprimo commented on subjects/java/projects/travel-plan/README.md:
```diff
@@ -0,0 +1,64 @@
+- Craft an Admin Dashboard enabling administrators to oversee users, travels, and payment gateways.
+- the admin should be able to add, edit or delete any user, travel, payment method and all the related part correctly (cascading!!!) 
+- Ensure each travel entry includes a destination or trajectory, dates, duration, activities or itinerary, accommodation, and transportation details.
+- for the payment gatways you can search about it to know what you need to support the most you have to support at least strip and paypal.
```
I would suggest renaming `payment gateways` to `payment methods` to be consistent with the first definition at line 36

nprimo commented on subjects/java/projects/travel-plan/README.md:
```diff
@@ -0,0 +1,64 @@
+- the admin should be able to add, edit or delete any user, travel, payment method and all the related part correctly (cascading!!!) 
+- Ensure each travel entry includes a destination or trajectory, dates, duration, activities or itinerary, accommodation, and transportation details.
+- for the payment gatways you can search about it to know what you need to support the most you have to support at least strip and paypal.
+- Develop an authentication and authorization service to safeguard access and operations.
```
I would add `... access and operations to the Admin Dashboard.` to avoid confusion and remark this is the only part of the bigger project the student are asked to implement for now

nprimo commented on subjects/java/projects/travel-plan/README.md:
```diff
@@ -0,0 +1,64 @@
+- Ensure each travel entry includes a destination or trajectory, dates, duration, activities or itinerary, accommodation, and transportation details.
+- for the payment gatways you can search about it to know what you need to support the most you have to support at least strip and paypal.
+- Develop an authentication and authorization service to safeguard access and operations.
+- Design a responsive and intuitive UI for the Admin Dashboard that shows well in diffirent screen sizes, ensuring browser compatibility at least for mozila and chrome.
```
I would put `Chrome` and `Mozilla` with Capital letter - not major

nprimo commented on subjects/java/projects/travel-plan/README.md:
```diff
@@ -0,0 +1,64 @@
+
+- Compose clear and detailed documentation for the application and database schema.
+- Incorporate Kubernetes alongside Ansible to enhance service management, orchestration, and load balancing capabilities.
+- impliment integration tests and E2E tests one of them or both of them. 
```
If this is a bonus, what is the CI/CD supposed to do? It is mentioned above:

```
Integrate a CI/CD pipeline via Jenkins for seamless build, test, and deployment workflows of PRs.
```

Should the CI/CD only be responsible to `build`? 
If the students are supposed to `deploy` should we give some places/services they should look at to deploy they're applications?

nprimo commented on subjects/java/projects/travel-plan/audit/README.md:
```diff
@@ -0,0 +1,80 @@
+
+##### Ask the student to describe the microservices architecture implemented.
+
+###### Did the architecture effectively split functionalities into different microservices based on key factors like scalability and domain separation?
```
Maybe add a high-level example of what we mean by this.
```
For example, there is a backend service responsible for reading information on the Postgres and Neo4j DB and providing information to a frontend service that displays the Admin Dashboard defined in the subject 
``` 
Or something similar - for example, are we expecting the final application to be behind a proxy webserver like Nginx or caddy?

nprimo commented on subjects/java/projects/travel-plan/audit/README.md:
```diff
@@ -0,0 +1,80 @@
+
+##### Discuss the CI/CD pipeline setup.
+
+###### is there unit tests for each functionality and are the test run for each PR.
```
The subject should reflect this request - as it is right now, tests seem to be only for bonus points. It might be better to specify that unit tests are mandatory

nprimo commented on subjects/java/projects/travel-plan/audit/README.md:
```diff
@@ -0,0 +1,80 @@
+
+###### is there unit tests for each functionality and are the test run for each PR.
+
+###### is SonarQube report free from any error or warning that can break the CI/CD Process.
```
This sentence should finish with a question mark

nprimo commented on subjects/java/projects/travel-plan/audit/README.md:
```diff
@@ -0,0 +1,80 @@
+
+##### Test each microservice API.
+
+###### Are all microservice APIs is role based and doesn't work with a admin role based user authenticated except the login api?
```
This point is not clear - is this supposed to be something like:
```
Are all API endpoints protected by authentication? Are all the endpoints only accessible to Admin users? (This last question might be not relevant for this project - unless the login/registration features are required for users as well. If this is the case, please, make it clearer in the subject)

zanninso replied:
when we ask them to implement authentication and authorization this include that they have to build a login page !!
okay i will apply the  change that you suggest looks better. thanks

nprimo replied:
This sentence is hard to understand for me. A possible rephrasing (but it might not be what you mentioned in the first place)

```
Is there a role-based management authentication implemented to access all the microservices' APIs? 
```

It is never mentioned in the subject to implement such a thing. 

Another option could be:
``
Are all the microservices' APIs only accessible when logged in with an Admin profile?
```

Or should be there some API (that displays data for example) available in general?

nprimo commented on subjects/java/projects/travel-plan/audit/README.md:
```diff
@@ -0,0 +1,80 @@
+
+###### Was the authentication service robust and did the role-based access control function correctly?
+
+##### Ask the student to Simulate load on microservices.
```
`... for example, by introducing some infinite loops in their API handlers.`

Or any other way you see might help to simulate extra load in the microservices

zanninso replied:
for this part is up to the students they should use there own way to prove that to the auditor

nprimo commented on subjects/java/projects/travel-plan/audit/README.md:
```diff
@@ -0,0 +1,80 @@
+
+###### Is the code consistence and well structured, also do they follow naming convention in recent pull requests?
+
+###### Are all the used libraries supported and not deprecated and not having any depracted methods in use, in recent pull requests?
```
Please, give an example on how to check for deprecated libraries and deprecated methods/functions

zanninso replied:
they have to take a lot at the Sonarqube  logs to see if there is i will add that to this question

nprimo replied:
I feel it would be nice to have this information in the audit to ensure that the person performing the audit knows what he/she should check :) For example
```
Check the Sonarqube logs to verify that there no error or warning related to the before mentioned potential issues
```

nprimo reviewed (changes requested):
I left some comments about changes that I feel are required to make the subject clearer.
Additionally, there are some minor grammar mistakes. I would recommend to use a plugin for grammar check in VSCode (or the editor you're using) or some CLI tools such as [write-good](https://github.com/btford/write-good) or [proselint](https://github.com/amperser/proselint)

zanninso commented on subjects/java/projects/travel-plan/README.md:
```diff
@@ -0,0 +1,64 @@
+- Craft an Admin Dashboard enabling administrators to oversee users, travels, and payment gateways.
+- the admin should be able to add, edit or delete any user, travel, payment method and all the related part correctly (cascading!!!) 
+- Ensure each travel entry includes a destination or trajectory, dates, duration, activities or itinerary, accommodation, and transportation details.
+- for the payment gatways you can search about it to know what you need to support the most you have to support at least strip and paypal.
```
no need they to do some efforts

zanninso commented on subjects/java/projects/travel-plan/audit/README.md:
```diff
@@ -0,0 +1,80 @@
+
+###### Did the microservices demonstrate effective load balancing and failover under heavy traffic?
+
+##### Verify Docker and Ansible setup.
```
yeah agree

nprimo commented on subjects/java/projects/travel-plan/README.md:
```diff
@@ -0,0 +1,66 @@
+#### 2. Development and Design
+
+- Craft an Admin Dashboard enabling administrators to oversee users, travels, and payment gateways.
+- the admin should be able to add, edit or delete any user, travel, payment method and all the related part correctly (database cascading!!!) 
```
I would change `(database cascading!!!)` with something more "neutral", for example:

````
(think about database cascading update and delete)
```

nprimo commented on subjects/java/projects/travel-plan/audit/README.md:
```diff
@@ -0,0 +1,80 @@
+
+##### Documentation Quality
+
+###### +did the students provides a clear documentation about the application and the database?
```
Missing capital letter at the beginning of the line

nprimo commented on subjects/java/projects/travel-plan/audit/README.md:
```diff
@@ -0,0 +1,80 @@
+
+##### Kubernetes Incorporation
+
+###### +did the students Incorporate Kubernetes alongside Ansible to enhance service management, orchestration, and load balancing capabilities?
```
Missing capital letter at the beginning of the line

nprimo commented on subjects/java/projects/travel-plan/audit/README.md:
```diff
@@ -0,0 +1,80 @@
+
+##### Additional Bonuses
+
+###### +did the student added any valuable bonuses and it works fine without any error 
```
Missing capital letter at the beginning of the line

nprimo commented on subjects/java/projects/travel-plan/audit/README.md:
```diff
@@ -0,0 +1,80 @@
+
+###### Is the code consistence and well structured, also do they follow naming convention in recent pull requests?
+
+###### Are all the used libraries supported and not deprecated and not having any depracted methods in use, in recent pull requests?
```
The second `deprecated` word is misspelt.

nprimo commented on subjects/java/projects/travel-plan/audit/README.md:
```diff
@@ -0,0 +1,80 @@
+
+##### Ask the student to explain one of the Ansible playbook
+
+###### did he/she clearly explain all the Ansible playbook
```
Missing capital letter at the beginning of the line

nprimo commented on subjects/java/projects/travel-plan/audit/README.md:
```diff
@@ -0,0 +1,80 @@
+
+##### Discuss the CI/CD pipeline setup.
+
+###### is there unit tests for each functionality and are the test run for each PR?
```
Missing capital letter at the beginning of the line

nprimo commented on subjects/java/projects/travel-plan/audit/README.md:
```diff
@@ -0,0 +1,80 @@
+
+###### is there unit tests for each functionality and are the test run for each PR?
+
+###### is SonarQube report free from any error or warning that can break the CI/CD Process?
```
Missing capital letter at the beginning of the line

nprimo commented on subjects/java/projects/travel-plan/README.md:
```diff
@@ -0,0 +1,66 @@
+- Compose clear and detailed documentation for the application and database schema.
+- Incorporate Kubernetes alongside Ansible to enhance service management, orchestration, and load balancing capabilities.
+- impliment integration tests and E2E tests one of them or both of them. 
+- any other bonus that add a real value to the project.
```
Missing capital letter at the beginning of the line

nprimo commented on subjects/java/projects/travel-plan/README.md:
```diff
@@ -0,0 +1,66 @@
+
+- Compose clear and detailed documentation for the application and database schema.
+- Incorporate Kubernetes alongside Ansible to enhance service management, orchestration, and load balancing capabilities.
+- impliment integration tests and E2E tests one of them or both of them. 
```
Missing capital letter at the beginning of the line

nprimo reviewed (changes requested):
As mentioned on Discord, please address the open conversation that does not have a reply or reaction linked to it to avoid stacking more comments/conversations on this PR

nprimo commented on subjects/java/projects/travel-plan/README.md:
```diff
@@ -0,0 +1,66 @@
+
+##### Use of Docker and Ansible:
+-    Docker to support automated provisioning and deployment of microservices.
+-    Ansible for creating playbooks to deploy all system elements consistently and scalably.
```
`scalably` is not a word - I would suggest to replace it with `in a scalable way`, for example

nprimo commented on subjects/java/projects/travel-plan/README.md:
```diff
@@ -0,0 +1,66 @@
+- Craft an Admin Dashboard enabling administrators to oversee users, travels, and payment gateways.
+- The admin should be able to add, edit or delete any user, travel, payment method and all the related parts correctly (think about database cascading update and delete).
+- Ensure each travel entry includes a destination or multiple destinations, dates, duration, activities, accommodation, and transportation details.
+- For the payment methods you can search about it to know what you need to support the most you have to support at least [Stripe](https://stripe.com/docs/development) and [PayPal](https://developer.paypal.com/home).
```
This sentence is not very clear to me. Could the following be enough or is there something else required?

```
The Admin Dashboard should support at least [Stripe](https://stripe.com/docs/development) and [PayPal](https://developer.paypal.com/home) as payment methods.
```

nprimo commented on subjects/java/projects/travel-plan/README.md:
```diff
@@ -0,0 +1,66 @@
+
+- Compose clear and detailed documentation for the application and database schema.
+- Incorporate Kubernetes alongside Ansible to enhance service management, orchestration, and load-balancing capabilities.
+- Implement integration tests and E2E tests one of them or both of them. 
```
I would rephrase this sentence as
```
Implement integration and/or E2E tests 
```

nprimo commented on subjects/java/projects/travel-plan/audit/README.md:
```diff
@@ -0,0 +1,98 @@
+
+###### Does your system maintain functionality even when one or more services fail?
+
+###### Is there an API Gateway in your architecture to manage incoming requests?
```
This is not explicitly required in the subject - are you considering this to be a "known" best practice to design a microservices application? Should this be added as a requirement in the subject?

For example, in section 1, `Configure a Microservices Architecture`
```
...
* Ensure microservices can only be accessed through an API Gateway 
```
Or something similar

zanninso replied:
a microservices system without a gateway is not a microservices system. so when i ask them to make microservices this automatically include a gateway to be present.

nprimo commented on subjects/java/projects/travel-plan/audit/README.md:
```diff
@@ -0,0 +1,98 @@
+
+###### Is there an API Gateway in your architecture to manage incoming requests?
+
+###### Can you track and trace a request across multiple services easily?
```
This point is not explicitly required in the subject. Should the subject mention that it is required to implement some sort of logging system to be able to track and trace requests across multiple services? Maybe this could be an additional point in section 1

nprimo commented on subjects/java/projects/travel-plan/audit/README.md:
```diff
@@ -0,0 +1,98 @@
+
+###### Can you track and trace a request across multiple services easily?
+
+###### Are communications between your microservices secured?
```
Nothing major, but I would add SSL/TLS protocol since they're mentioned in the subject.

zanninso replied:
i don't see that i should mention that here since it's already mentioned at line 36

nprimo replied:
Should this question be removed then, since it is asked at line 36 the same thing?

nprimo commented on subjects/java/projects/travel-plan/audit/README.md:
```diff
@@ -0,0 +1,98 @@
+
+##### Discuss the CI/CD pipeline setup.
+
+###### Are there unit tests for each functionality and are the test runs for each PR?
```
I would change the second part of the sentence as
```
... and are the tests running for each new PR?
```

nprimo commented on subjects/java/projects/travel-plan/audit/README.md:
```diff
@@ -0,0 +1,98 @@
+
+###### Are all the microservices' APIs only accessible when logged in with an Admin profile?
+
+###### Are all microservice APIs functioning correctly and handling errors effectively?
```
I feel this point is already covered in the two points at lines 58 and 60. I would remove this line and keep the other two

nprimo commented on subjects/java/projects/travel-plan/audit/README.md:
```diff
@@ -0,0 +1,98 @@
+
+###### Did the microservices demonstrate effective load balancing and failover under heavy traffic?
+
+##### Verify Docker and Ansible setup.
```
Shouldn't this point be moved after the section `Verify the execution of Ansible playbooks`?
Are Docker containers supposed to be started by the Ansible playbooks?

zanninso replied:
yes docker docker container supposed to be started by Ansible playbooks

nprimo commented on subjects/java/projects/travel-plan/audit/README.md:
```diff
@@ -0,0 +1,98 @@
+
+##### Assess code review and best practices.
+
+###### Is the code consistence and well structured, also do they follow naming conventions in recent pull requests?
```
I would split this point into two separate ones if I understood what you're asking properly.

1. Is the code consistent and well-structured?
2. Are all pull requests following naming conventions? (Add examples of what you mean here)

nprimo replied:
I don't see any changes linked to this comment - am I missing something? :sweat_smile:

zanninso replied:
look at lines 76 and 78

nprimo replied:
I would remove line 80 since it's a repetition at this point.

nprimo commented on subjects/java/projects/travel-plan/audit/README.md:
```diff
@@ -0,0 +1,98 @@
+
+###### Is the code consistence and well structured, also do they follow naming conventions in recent pull requests?
+
+###### Are all the used libraries supported and not deprecated and not having any deprecated methods in use, in recent pull requests check SonarQube?
```
I would split this sentence to make it easier to read and understand.
For example:

```
Check SonarQube logs in recent pull requests: is the log free of warnings about unsupported or deprecated libraries? 
```

nprimo reviewed (changes requested):
I left some comments in line to improve the clarity of some parts of the subject and audit. 

I found two audit questions that ask for specific features that are not clearly defined in the subject (API Gateway and Logging system).

Additionally, it is mentioned to use the least privilege principle as a security measure and to use role-based access controls: isn't this supposed to be only for admin? Is it expected to have more than one level of access to the platform? This could create confusion in the audit phase of the project.

nprimo commented on subjects/java/projects/travel-plan/audit/README.md:
```diff
@@ -79,7 +77,9 @@
-###### Are all the used libraries supported and not deprecated and not having any deprecated methods in use, in recent pull requests check SonarQube?
+##### Check SonarQube logs in recent pull requests.
+
+###### is the log free of warnings about unsupported or deprecated libraries?
```
Should start with a capital letter

nprimo commented on subjects/java/projects/travel-plan/audit/README.md:
```diff
@@ -0,0 +1,100 @@
+
+##### Ask the student to explain one of the Ansible playbook
+
+###### Did he/she clearly explain all the Ansible playbook
```
There is a missing question mark at the end of this line

nprimo reviewed (changes requested):
I have added two minor change requests inline.

@zanninso  it is mentioned to use the least privilege principle as a security measure and to use role-based access controls: isn't this supposed to be only for admin? Is it expected to have more than one level of access to the platform? This could create confusion in the audit phase of the project.

nprimo commented:
@zanninso as I have pointed out on Discord, could you please reply to [this](https://github.com/01-edu/public/pull/2282#pullrequestreview-1796204024)? As soon as this comment is addressed, we can approve and merge this PR :)

zanninso commented:
> I have added two minor change requests inline.
> 
> @zanninso it is mentioned to use the least privilege principle as a security measure and to use role-based access controls: isn't this supposed to be only for admin? Is it expected to have more than one level of access to the platform? This could create confusion in the audit phase of the project.



> I have added two minor change requests inline.
> 
> @zanninso it is mentioned to use the least privilege principle as a security measure and to use role-based access controls: isn't this supposed to be only for admin? Is it expected to have more than one level of access to the platform? This could create confusion in the audit phase of the project.

for this part of the project they will have just the admin role but in the final project they have to add more roles, it's already explained in subject. i don;t see that will make any confusion

nprimo reviewed (approved):
LGTM
